api/storage/client.go

```diff
@@ -147,3 +147,16 @@ func (c *Client) AddToUnit(storages []params.StorageAddParams) ([]params.ErrorRe
 	}
 	return out.Results, nil
 }
+
+// Destroy destroys specified storage entities.
+func (c *Client) Destroy(tags []names.Tag) ([]params.ErrorResult, error) {
```

Comment:
Sadly we have not enforced this across the board, but the original design was that API callers would not deal with tags but rather entities and natural keys etc. The tags were meant to be an on the wire thing only, internal to the api / apiserver layers. The abstraction has leaked in various places :-( Would be good not to leak here too considering the ListFileSystems() APi on this same facade does it the right way, ie takes natural keys as arguments and converts to tags.

The CLI takes user facing args and converts to tags so my thinking was similar logic can go in the API layer, just like for ListFileSystems().